src/Notification/NotificationClientFactory.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace OpenEuropa\EPoetry\Notification;
+
+use Http\Client\HttpClient;
+use Phpro\SoapClient\ClientFactory as PhproClientFactory;
+use Phpro\SoapClient\ClientBuilder;
+use Phpro\SoapClient\ClientFactory;
+use OpenEuropa\EPoetry\Services\LoggerDecorator;
+use OpenEuropa\EPoetry\Services\LoggerSubscriber;
+use Phpro\SoapClient\Soap\Handler\HttPlugHandle;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+
+class NotificationClientFactory
+{
+    /**
+     * Event dispatcher instance.
+     *
+     * @var \Symfony\Component\EventDispatcher\EventDispatcherInterface
+     */
+    protected $eventDispatcher;
+
+    /**
+     * HTTP Client.
+     *
+     * @var \Http\Client\HttpClient
+     */
+    protected $httpClient;
+
+    /**
+     * PSR3 logger instance.
+     *
+     * @var \Psr\Log\LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * Level of severity of the events to be logged.
+     *
+     * @var string
+     */
+    protected $logLevel = 'none';
+
+    /**
+     * List of Phpro\SoapClient middlewares.
+     *
+     * @var \Phpro\SoapClient\Middleware\MiddlewareInterface[]
+     */
+    protected $middlewares = [];
+
+    /**
+     * PHP SOAP client options.
+     *
+     * @see http://php.net/manual/en/soapclient.soapclient.php
+     *
+     * @var array
+     */
+    protected $options = [];
+
+    /**
+     * URI of the WSDL file.
+     *
+     * @var string
+     */
+    protected $wsdl;
+
+    /**
+     * Factory constructor.
+     */
+    public function __construct(string $wsdl, HttpClient $httpClient, array $options = [])
+    {
+        $this->wsdl = $wsdl;
+        $this->httpClient = $httpClient;
+        $this->options = [
+            'cache_wsdl' => WSDL_CACHE_NONE,
+        ] + $options;
+    }
+
+    public static function factory(string $wsdl): NotificationClient
+    {
+        $clientFactory = new PhproClientFactory(NotificationClient::class);
+        $clientBuilder = new ClientBuilder($clientFactory, $wsdl, []);
+        $clientBuilder->withClassMaps(NotificationClassmap::getCollection());
+
+        return $clientBuilder->build();
+    }
+
+    /**
+     * Construct and return an ePoetry client instance.
+     *
+     * Make sure you set/add logger, event dispatcher and middlewares before
```

Comment:
There are no methods for setting any of this parameters.

Reply:
Methods added.